### PR TITLE
ProviderManager should have a varargs constructor

### PR DESCRIPTION
**Pull request by ThomasVitale**
- Added varargs constructor to ProviderManager.
- Updated ProviderManagerTests to use the new syntax and fix the check for empty list.

Fixes gh-7713

### CODE REVIEW

File: core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java [hunk 1]
```diff
@@ -96,7 +94,7 @@ public void authenticationSucceedsWhenFirstProviderReturnsNullButSecondAuthentic
 
 	@Test(expected = IllegalArgumentException.class)
 	public void testStartupFailsIfProvidersNotSet() {
-		new ProviderManager(null);
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java]: I think we should still have a test for `new ProviderManager((AuthenticationProvider) null)` and for `new ProviderManager((List<AuthenticationProvider>) null)`.

The reason is that an application might do:

```java
AuthenticationProvider provider = constructProvider();
return new ProviderManager(provider);
```

but `constructProvider` returns `null`.

I believe this also implies that the extra validation logic in `checkState()` will turn out to be necessary.

File: core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java [hunk 2]
```diff
@@ -189,7 +186,7 @@ public void parentAuthenticationIsUsedIfProvidersDontAuthenticate() {
 		Authentication authReq = mock(Authentication.class);
 		when(parent.authenticate(authReq)).thenReturn(authReq);
 		ProviderManager mgr = new ProviderManager(
-				Arrays.asList(mock(AuthenticationProvider.class)), parent);
+				Collections.singletonList(mock(AuthenticationProvider.class)), parent);
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java]: Thanks for these cleanup items, too. Will you please put them in a separate commit (same PR)? That is, any code changes that aren't directly related to the new constructor, let's separate them into their own commit. This simplifies maintenance.

**Comment by jzheaux** [comment]
@ThomasVitale Will you be able to apply the requested changes?

File: core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java (hunk 1 above)
Comment (ThomasVitale) [reply core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java]: Got it, thanks for the clarification. I have added the extra validation login in `checkState()` and tests for both null scenarios.

File: core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java (hunk 2 above)
Comment (ThomasVitale) [reply core/src/test/java/org/springframework/security/authentication/ProviderManagerTests.java]: Ok, now they are in their own commit.

**Comment by ThomasVitale** [comment]
@jzheaux I have just delivered the requested changes, sorry for the delay. Does it look ok now?

**Comment by jzheaux** [comment]
Nice, @ThomasVitale! This is now merged into `master`.

Also, I added a very slight polish via d22b476983d476b46e2ad97fcad7cd59f650ad5d
